baseplate/clients/redis.py

```diff
@@ -269,7 +269,39 @@ def __init__(
     # pylint: disable=arguments-differ
     def execute(self, **kwargs: Any) -> Any:
         with self.server_span.make_child(self.trace_name):
-            return super().execute(**kwargs)
+
+            success = "true"
+            start_time = perf_counter()
+            labels = {
+                f"{PROM_LABELS_PREFIX}_command": "pipeline",
+                f"{PROM_LABELS_PREFIX}_client_name": self.connection_pool.connection_kwargs.get(
+                    "client_name", ""
+                ),
+                f"{PROM_LABELS_PREFIX}_database": self.connection_pool.connection_kwargs.get(
+                    "db", ""
+                ),
+                f"{PROM_LABELS_PREFIX}_type": "standalone",
+            }
+            num_reqs = len(self.command_stack)
+            commands = [args[0] for (args, options) in self.command_stack]
+
+            try:
+                with ACTIVE_REQUESTS.labels(**labels).track_inprogress():
+                    return super().execute(**kwargs)
+            except:  # noqa: E722
+                success = "false"
+                raise
+            finally:
+                for command in commands:
```

Comment:
Since no one's commented on it: This has a linear performance impact with the number of commands in the pipeline. It's probably not a noticeable impact, but an option here is to deviate from the spec and drop the `command` label. The call would then just become a singular `.inc(len(self.command_stack))`.